### ISSUE TITLE
Potential memory leak with gzip reader

### ISSUE THREAD

**Issue by tleyden** [couchbase/sync_gateway#953]
https://github.com/couchbase/sync_gateway/blob/master/src/github.com/couchbase/sync_gateway/db/attachment.go#L162

```
if input, err = gzip.NewReader(input); err != nil {…}
```

the docs for gunzip.NewReader say:

> It is the caller's responsibility to call Close on the Reader when done.  

I’m not seeing where we are closing that reader returned from NewReader.

Comments from @adamcfraser:

> I agree that this needs attention.  it's hard to follow the way we're turning an io.Reader into a gzip.Reader - where the gzip.Reader expects a close, and the io.Reader doesn't

**Comment by adamcfraser** [comment]
Moved to https://issues.couchbase.com/browse/CBG-645
